File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.github.pagehelper.PageInfo;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.ResponseResult;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {
    @Reference
    private BrandService brandService;

    @RequestMapping("/findAll")
    public List<TbBrand> findAll() {
        System.out.println("========================");
        return brandService.findAll();
    }
    /*
     * //分页插件4.0.0做法
     *
     * @RequestMapping("/findPage") public PageResult findPage(Integer
     * pageNum,Integer pageSize) { return brandService.findPage(pageNum,pageSize); }
     */

    // 分页插件5.1.2做法 数据只有一个list,错误写法
    @RequestMapping("/findPage1")
    public PageInfo findPage1(Integer pageNum, Integer pageSize) {
        List<TbBrand> tbBrandList = brandService.findPage1(pageNum, pageSize);
        return new PageInfo<>(tbBrandList);
    }
    // 分页插件5.1.2 service层包装数据ok
    @RequestMapping("/findPage")
    public PageInfo findPage(Integer pageNum, Integer pageSize) {
        return brandService.findPage(pageNum, pageSize);
    }

    @RequestMapping("/add")
    public ResponseResult add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);
            return new ResponseResult(true, "新增成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseResult(false, "新增失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findOne(id);
    }
    @RequestMapping("/update")
    public ResponseResult update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new ResponseResult(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseResult(false, "修改失败");
        }
    }
    @RequestMapping("/delBrand")
    public ResponseResult delBrand(Long[] ids){
        try {
            brandService.delBrand(ids);
            return new ResponseResult(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new ResponseResult(false, "删除失败");
        }
    }
    @RequestMapping("/searchPage")
    public PageInfo searchPage(@RequestBody TbBrand brand, Integer pageNum, Integer pageSize) {
        if (brand == null) {
            return brandService.findPage(pageNum, pageSize);
        }
        return brandService.searchPage(brand, pageNum, pageSize);
    }

    /**
     * 查询所有的品牌[{"id": 1, "text": "联想"}, {"id": 3, "text": "三星"}]
     * @return 返回select2下拉列表格式的数据形式
     */
    @RequestMapping("/select2BrandList")
    public List<Map> select2BrandList(){
        return brandService.select2BrandList();
    }
}

File: pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller(
    "brandController",
    function ($scope, $controller, brandService) {
        $controller("baseController", {$scope: $scope});//继承
        //读取列表数据绑定到表单
        $scope.findAll = function () {
            brandService.findAll().success(
                function (response) {
                    $scope.brandList = response;
                })
        };
        //分页  no.3  使用条件查询以后就不需要这个方法了
        $scope.findPage = function (pageNum, pageSize) {
            brandService.findPage(pageNum, pageSize).success(
                function (data) {
                    $scope.list = data.list;
                    $scope.paginationConf.totalItems = data.total;
                })
        };
        //根据id查询后修改
        $scope.findOne = function (id) {
            brandService.findOne(id).success(function (data) {
                $scope.entity = data;
            })
        };
        //新增与修改
        $scope.save = function () {
            var serviceObject;//服务层对象
            if ($scope.entity.id) {
                serviceObject = brandService.update($scope.entity);
            } else {
                serviceObject = brandService.add($scope.entity);
            }
            serviceObject.success(function (data) {
                if (data.success) {
                    $scope.reloadList();
                } else {
                    alert(data.message)
                }
            });
        };
        //批量删除
        $scope.del = function () {
            if (confirm("您确定删除吗?")) {
                brandService.del($scope.ids).success(function (data) {
                    if (data.success) {
                        $scope.ids = [];
                        $scope.reloadList();
                    } else {
                        alert(data.message)
                    }
                });
            }
        };
        $scope.list = [];//每页的数据
        //搜索
        $scope.searchPage = function (pageNum, pageSize) {
            brandService.searchPage(pageNum, pageSize, $scope.searchEntity).success(function (data) {
                $scope.list = data.list; //定义分页每页的数据
                $scope.paginationConf.totalItems = data.total;
            })
        }
    });
